From e2ecefa3c3c7925f66d70dc4ef5d17d1e9b27840 Mon Sep 17 00:00:00 2001
From: Richard Hughes <richard@hughsie.com>
Date: Tue, 23 May 2017 13:54:09 +0100
Subject: [PATCH 3/3] unifying: Relax the requirement that a SHORT request
 should return a SHORT reply

For some HID++2.0 devices a SHORT request is allowed to be padded into a LONG
response.
---
 plugins/unifying/lu-device.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/plugins/unifying/lu-device.c b/plugins/unifying/lu-device.c
index ac3e68b..4e53633 100644
--- a/plugins/unifying/lu-device.c
+++ b/plugins/unifying/lu-device.c
@@ -288,13 +288,13 @@ lu_device_hidpp_receive (LuDevice *device,
 		}
 	}
 
-	/* check length */
+	/* check long enough, but allow returning oversize packets */
 	lu_device_hidpp_dump (device, "device->host", (guint8 *) msg, read_size);
-	if (lu_device_hidpp_msg_length (msg) != read_size) {
+	if (read_size < lu_device_hidpp_msg_length (msg)) {
 		g_set_error (error,
 			     G_IO_ERROR,
 			     G_IO_ERROR_FAILED,
-			     "not expected message length, "
+			     "message length too small, "
 			     "got %" G_GSIZE_FORMAT " expected %" G_GSIZE_FORMAT,
 			     read_size, lu_device_hidpp_msg_length (msg));
 		return FALSE;
-- 
2.7.4

